src/lib.rs

```diff
@@ -120,15 +120,44 @@
 //!     assert_eq!(DonkeyKong::new(3), -&dk);
 //! }
 //! ```
+//! ## Generics
+//! Any of the above forms can additionally be generic by putting generic params just before the lambda.
+//! ```
+//! // impl_op!(op $[attr] <generic_params> |a: LHS, b: RHS| -> OUT {...});
+//! // where
+//! // generic_params: anything that would go just after an `impl`, including lifetimes and
+//! //                 joining multiple bounds with `+`
+//!
+//! use auto_ops::impl_op;
+//!
+//! #[derive(Debug, PartialEq)]
+//! struct Barrel<T> {
+//!     pub bananas: T,
+//! }
+//!
+//! impl<T> Barrel<T> {
+//!     pub fn new(bananas: T) -> Self {
+//!         Barrel { bananas }
+//!     }
+//! }
+//!
+//! impl_op!(+ <T: ::std::ops::Add<T>> |a: Barrel<T>, b: Barrel<T>| -> Barrel<<T as ::std::ops::Add<T>>::Output> {
+//!     Barrel::new(a.bananas + b.bananas)
+//! });
+//!
+//! assert_eq!(Barrel::new(5u8), Barrel::new(2u8) + Barrel::new(3u8));
+//! assert_eq!(Barrel::new(1.2f32), Barrel::new(0.5f32) + Barrel::new(0.7f32));
+//! ```
```

Comment:
Wanted to double check some stuff with these doc updates, broadly with the question of "do they generally look okay?", and specifically some things I've thought about:
- Should the previous examples/docs be updated to include `<generic_params>` in them?
- Should the example be a bit more sophisticated?
- Should it explicitly describe what's going on as opposed to just relying on this one example?

I'm not necessarily looking for detailed answers on every count if that doesn't feel useful, just some questions to spark ideas!